web/html/src/manager/minion/proxy/proxy-config.tsx

```diff
@@ -0,0 +1,527 @@
+import * as React from "react";
+import { useEffect, useState } from "react";
+
+import { AsyncButton, SubmitButton } from "components/buttons";
+import { Select } from "components/input";
+import { Form } from "components/input/form/Form";
+import { FormMultiInput } from "components/input/form-multi-input/FormMultiInput";
+import { unflattenModel } from "components/input/form-utils";
+import { Radio } from "components/input/radio/Radio";
+import { Text } from "components/input/text/Text";
+import { Messages } from "components/messages/messages";
+import { TopPanel } from "components/panels/TopPanel";
+import Validation from "components/validation";
+
+import { useDebounce } from "utils/hooks";
+import Network from "utils/network";
+
+import { ContainerConfigMessages } from "./proxy-config-messages";
+import {
+  ProxyConfigModel,
+  ProxyConfigProps,
+  RegistryMode,
+  SourceMode,
+  TagOptions,
+  UseCertsMode,
+} from "./proxy-config-types";
+import {
+  getRegistryData,
+  imageNames,
+  modelDefaults,
+  readFileFields,
+  restoreRegistryInputs,
+  retrieveRegistryTags,
+} from "./proxy-config-utils";
+
+export function ProxyConfig({
+  serverId,
+  isUyuni,
```

Comment:
Recommend using `import { isUyuni } from "core/user-preferences";` instead, we have it globally available since so many places need this information.